pkg/docker/docker.go

```diff
@@ -452,58 +558,77 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 }
 
 // CheckImage checks image from the local registry.
-func (d *stiDocker) CheckImage(name string) (*docker.Image, error) {
+func (d *stiDocker) CheckImage(name string) (*api.Image, error) {
 	name = getImageName(name)
-	image, err := d.client.InspectImage(name)
+	inspect, err := d.kubeDockerClient.InspectImage(name)
 	if err != nil {
 		glog.V(4).Infof("error inspecting image %s: %v", name, err)
 		return nil, errors.NewInspectImageError(name, err)
 	}
-	return image, nil
+	if inspect != nil {
+		image := api.Image{}
+		updateImageWithInspect(&image, inspect)
+		return &image, nil
+	}
+
+	return nil, nil
 }
 
 // PullImage pulls an image into the local registry
-func (d *stiDocker) PullImage(name string) (*docker.Image, error) {
+func (d *stiDocker) PullImage(name string) (*api.Image, error) {
 	name = getImageName(name)
-	glog.V(1).Infof("Pulling Docker image %s ...", name)
-	// TODO: Add authentication support
-	if err := d.client.PullImage(docker.PullImageOptions{Repository: name}, d.pullAuth); err != nil {
-		glog.V(3).Infof("An error was received from the PullImage call: %v", err)
+	err := d.kubeDockerClient.PullImage(name, d.pullAuth, dockertypes.ImagePullOptions{})
+	if err != nil {
 		return nil, errors.NewPullImageError(name, err)
 	}
-	image, err := d.client.InspectImage(name)
+	resp, err := d.kubeDockerClient.InspectImage(name)
 	if err != nil {
-		glog.V(4).Infof("error inspecting image %s: %v", name, err)
-		return nil, errors.NewInspectImageError(name, err)
+		return nil, errors.NewPullImageError(name, err)
+	}
+	if resp != nil {
+		image := api.Image{}
+		updateImageWithInspect(&image, resp)
+		return &image, nil
+	}
+	return nil, nil
+}
+
+func updateImageWithInspect(image *api.Image, inspect *dockertypes.ImageInspect) {
+	image.ID = inspect.ID
+	if inspect.Config != nil {
+		image.ConfigLabels = inspect.Config.Labels
+		image.ConfigEnv = inspect.Config.Env
+	}
+	if inspect.ContainerConfig != nil {
+		image.ContainerLabels = inspect.ContainerConfig.Labels
+		image.ContainerEnv = inspect.ContainerConfig.Env
```

Comment:
would it make sense to more closely align the api.Image struct with the docker image struct?